bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/service/impl/BahmniFormDetailsServiceImplTest.java

```diff
@@ -0,0 +1,367 @@
+package org.bahmni.module.bahmnicore.service.impl;
+
+import org.bahmni.module.bahmnicore.forms2.contract.form.FormType;
+import org.bahmni.module.bahmnicore.forms2.contract.form.data.FormDetails;
+import org.bahmni.module.bahmnicore.forms2.mapper.FormDetailsMapper;
+import org.bahmni.module.bahmnicore.forms2.service.impl.BahmniFormDetailsServiceImpl;
+import org.bahmni.module.bahmnicore.forms2.util.Form2ObsUtil;
+import org.bahmni.module.bahmnicore.service.BahmniProgramWorkflowService;
+import org.bahmni.module.bahmnicore.service.BahmniVisitService;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.Visit;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.parameter.EncounterSearchCriteria;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.security.InvalidParameterException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyListOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+@PrepareForTest({FormType.class, Form2ObsUtil.class, FormDetailsMapper.class})
+@RunWith(PowerMockRunner.class)
+public class BahmniFormDetailsServiceImplTest {
```

Comment:
IMO, there is unnecessary mocking of static classes FormUtil, even the enum FormType is mocked.

Reply:
Will remove enum mocks and keep the static method calls.